be/src/olap/merger.cpp

```diff
@@ -257,6 +274,11 @@ Status Merger::vertical_compact_one_group(
         output_rows += block.rows();
         block.clear_column_data();
     }
+    if (is_key && reader_params.record_rowids > 0) {
+        RETURN_IF_ERROR(stats_output->rowid_conversion->save_to_file());
```

Comment:
We can add a configuration to control persistence only when the memory used for rowid conversion exceeds the threshold.

Reply:
+1